app/main.py

```diff
@@ -1,3 +1,45 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, new: int | float | Distance) -> Distance:
+        if isinstance(new, (int, float)):
+            return Distance(self.km + new)
+        return Distance(self.km + new.km)
+
+    def __iadd__(self, new: int | float | Distance) -> None:
+        if isinstance(new, (int, float)):
+            self.km += new
+        else:
+            self.km += new.km
+        return self
+
+    def __mul__(self, new: int) -> Distance:
+        return Distance(self.km * new)
+
+    def __truediv__(self, new: int) -> Distance:
+        return Distance(round(self.km / new, 2))
+
+    def __eq__(self, new: int | Distance) -> bool:
+        return self.km == new
```

Comment:
What if new is of type Distance here and in all the following methods?
Yurii already commented you about this

Reply:
I can make it more complex by checking `isinstance(new, Distance)` but why?

Reply:
or maybe make another method to convert in one type but still why